contractrequester/contractrequester.go

```diff
@@ -118,16 +119,25 @@ func (cr *ContractRequester) Call(ctx context.Context, inMsg insolar.Message) (i
 	}
 	msg.Sender = cr.JetCoordinator.Me()
 
-	var seq uint64
+	var reqHash [insolar.RecordHashSize]byte
 	var ch chan *message.ReturnResults
 
 	if !async {
 		cr.ResultMutex.Lock()
-		cr.Sequence++
-		seq = cr.Sequence
-		msg.Sequence = seq
+		virtRec := record.Wrap(msg.IncomingRequest)
+		buf, err := virtRec.Marshal()
+		if err != nil {
+			return nil, errors.Wrap(err, "[ ContractRequester::Call ] Failed to marshal record")
+		}
+		h := cr.PCS.ReferenceHasher()
+		_, err = h.Write(buf)
+		if err != nil {
+			return nil, errors.Wrap(err, "[ ContractRequester::Call ] Failed to calculate hash")
+		}
+
+		copy(reqHash[:], h.Sum(nil)[0:insolar.RecordHashSize])
```

Comment:
Can we make this a function? A private function within ContractRequester, but function.

Reply:
ok